### PR TITLE
Added AdaLAM wrapper.

### PR DESCRIPTION
**Pull request by mihaidusmanu**
## Some results on South-Building

SIFT + MNN
```
    num_reg_images = 128
    num_cameras = 1
    num_points3D = 40544
    num_observations = 215421
    mean_track_length = 5.31326
    mean_observations_per_image = 1682.98
    mean_reprojection_error = 0.930621
    num_input_images = 128
```

SIFT + AdaLAM
```
   num_reg_images = 128
   num_cameras = 1
   num_points3D = 38752
   num_observations = 214480
   mean_track_length = 5.53468
   mean_observations_per_image = 1675.62
   mean_reprojection_error = 0.949959
   num_input_images = 128
```

## Extra
- Bugfix in GV if image has no keypoints

## TODO
- Add  support for features without ori / scale (by setting `orientation_difference_threshold` and `scale_rate_threshold` to `None`)

### CODE REVIEW

**Comment by mihaidusmanu** [comment]
@cavalli1234 for double checking

**Review by cavalli1234 — APPROVED** [review]
Looks all good!

- Just consider that, in case confidence might be useful, an estimate of it can be extracted from AdaLAM (in adalam/core.py, function adalam_core, variable conf). It is the ratio between the local inliers found divided by the expected number of random false positives you would get from outliers only in the same local neighborhood, so 1 means insignificant, and goes up unbounded.

File: hloc/extractors/dog.py [hunk 1]
```diff
@@ -94,18 +96,24 @@ def _forward(self, data):
...
             indices = torch.topk(scores, self.conf['max_keypoints'])
             keypoints = keypoints[indices]
+            scales = scales[indices]
+            oris = oris[indices]
             scores = scores[indices]
             descriptors = descriptors[indices]
 
         return {
             'keypoints': keypoints[None],
+            'scales': scales[None],
```
Comment (Skydes) [inline hloc/extractors/dog.py:115]: When the input image is downsized, don't you need to also modify the scale in `extract_features.py` to account for it?

Comment (mihaidusmanu) [reply hloc/extractors/dog.py:115]: Good catch, I'll fix it!

**Pull request merged**